### PR TITLE
Update /actions page #75 (Finish): Added in share.

### PR DESCRIPTION
**Pull request by MazharulIslam-Naim**
## Overview 
This pr adds sharing about actions taken by the player on their social media accounts. It mainly focuses on getting the preview for both invite and share to work. 

## Test Plan 
This was tested manually on Desktop and iPhone.

## Follow ups 
The playerwelcome page in preview might not exactly look how the page looks regularly, so let me know if that needs tweaking.

### CODE REVIEW

**Review by Devanshusp — CHANGES_REQUESTED** [review]
Hey Naim, thanks for working on this pr!

This is so cool! I thought the preview was just supposed to be an image, this is much more robust. One nit through, I would like to see the preview in one page without the first scroll bar.  Feel free to adjust the size of the phone preview.
image

Also, if we could relatively decrease the size of the page (fonts, buttons, icons), I think it would look even better. Something like the figma:
image

**Comment by MazharulIslam-Naim** [comment]
Just leave this styling issue for beta. I have noted it down and I'll fix it later.

**Comment by rickyyx** [comment]
Hey Naim, I think there are 2 issues I encountered locally that need to be fixed before merging this

1. It is throwing errors when the players get the link
Steps to reproduce:
- Create a challenger, share the link 
- Open an Incognito window, click on the link 
- While it was being loaded on `/actions` page, it crashed with the below: 
image

I believe the issue is there is no `currentUser` logged in yet, so we could not `getUserDatabase()` here in `Invite.js` 


2. Looks like changing the feedback from `<a>` tag to `<button>` creates some styling issue: 
image

Does it show up on your end?

File: src/components/Footer.js [hunk 1]
```diff
@@ -6,23 +6,24 @@ import {
   AiOutlineInstagram,
 } from "react-icons/ai";
 
-function Footer() {
+function Footer({ isShare }) {
+  // If isShare is not true then the social media buttons work.
```
Comment (rickyyx) [inline src/components/Footer.js:10]: Why does `isShare` related to the buttons here?

Comment (rickyyx) [reply src/components/Footer.js:10]: oh, coz of the sharing preview... hmm I see. This is pretty hacky.

**Review by rickyyx — APPROVED** [review]
This is actually a rather complex PR, so well done @MazharulIslam-Naim for putting this up in the first place. 

I am not too sure about the additional `isShare` check everyone to get the `preview` thing working is really worthwhile though. Didn't know we have to do it in such a hacky way, or I would have asked the design team to change the preview to just a static image or even get rid of this requirement. 

But it seems to be working - I tested the local share with both players and challengers, and the non-share views. So why not then, I don't want this little magic get wasted... And anyway,  there is documentation to explain why the `isShare`.

**Review by rickyyx — APPROVED** [review]

**Pull request merged**